tests/scripts/components-build-system.sh

```diff
@@ -135,7 +135,32 @@ component_test_cmake_as_package () {
     make
     ./cmake_package
     if [[ "$OSTYPE" == linux* ]]; then
-        PKG_CONFIG_PATH="${build_variant_dir}/mbedtls/pkgconfig" ${root_dir}/tests/scripts/pkgconfig.sh
+        PKG_CONFIG_PATH="${build_variant_dir}/mbedtls/pkgconfig" \
+        ${root_dir}/tests/scripts/pkgconfig.sh \
+        mbedtls mbedx509 mbedcrypto
+        # These are the EXPECTED package names. Renaming these could break
+        # consumers of pkg-config, consider carefully.
+    fi
+}
+
+component_test_tf_psa_crypto_cmake_as_package () {
+    # Remove existing generated files so that we use the ones CMake
+    # generates
+    make neat
+
+    msg "build: cmake 'as-package' build"
+    root_dir="$(pwd)"
+    cd tf-psa-crypto/programs/test/cmake_package
+    build_variant_dir="$(pwd)"
```

Comment:
Not a blocker

Having two distinct variablles invoke pwd looks like funcional redudancy.  I do understand that in the future we will want to have `build_variant_dir != root_dir` but we could for now at least copy the strings?

`build_variant_dir = "${root_dir}`

Reply:
We have `build_variant_dir != root_dir` as we change the current directory between the two usages of `$(pwd)`.